index.html

```diff
@@ -0,0 +1,231 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/all.min.css" 
+    integrity="sha512-SzlrxWUlpfuzQ+pcUCosxcglQRNAq/DZjVsC0lE40xsADsfeQoEypE+enwcOiGjk/bSuGGKHEyjSoQ1zVisanQ==" crossorigin="anonymous" referrerpolicy="no-referrer" />
+    <style>
+      body {
+  background-color: #f8f8f8;
+}
+
+.container {
+  background: #fff;
+  width: 50%;
+  height: 70%;
+  margin: auto;
+  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
+}
+
+.header {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  padding-right: 8px;
+}
+
+.head-btn,
+.btn,
+.input-btn {
+  border: none;
+  background: #fff;
+  cursor: pointer;
+}
+
+.input-field {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  padding-right: 8px;
+  height: 20px;
+}
+
+hr {
+  height: 2px;
+  border-width: 0;
+  color: gray;
+  background-color: #ddd;
+}
+
+.todo-list {
+  padding: 0;
+}
+
+.todo-list li {
+  list-style: none;
+  line-height: 0.5rem;
+}
+
+.input-btn {
+  float: right;
+  font-size: 16px;
+  padding-right: 20px;
+}
+
+.title,
+.input {
+  border: none;
+  margin-left: 44px;
+}
+
+.delete {
+  float: right;
+  margin-right: 12px;
+  cursor: pointer;
+}
+
+.check {
+  text-decoration: line-through;
+}
+
+.clr-btn {
+  width: 100%;
+  cursor: pointer;
+  border: none;
+}
+    </style>
```

Comment:
- [ ] As per the requiremnt, to follow HTML & CSS best practices, kindly create a new file for the CSS part.